src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeUpdateTribunalDecisionRule31PersonalisationEmailTest.java

```diff
@@ -0,0 +1,182 @@
+    @Mock
+    AppealService appealService;
+    private LegalRepresentativeUpdateTribunalDecisionRule31PersonalisationEmail legalRepresentativeUpdateTribunalDecisionRule31PersonalisationEmail;
+
+    @BeforeEach
+    public void setup() {
+        when(callback.getCaseDetails()).thenReturn(caseDetails);
+        when(caseDetails.getCaseData()).thenReturn(asylumCase);
+        when(callback.getCaseDetails().getId()).thenReturn(caseId);
+
+        when(asylumCase.read(APPEAL_REFERENCE_NUMBER, String.class)).thenReturn(Optional.of(appealReferenceNumber));
+        when(asylumCase.read(LEGAL_REP_REFERENCE_NUMBER, String.class)).thenReturn(Optional.of(legalRepReferenceNumber));
+        when(asylumCase.read(APPELLANT_GIVEN_NAMES, String.class)).thenReturn(Optional.of(appellantGivenNames));
+        when(asylumCase.read(APPELLANT_FAMILY_NAME, String.class)).thenReturn(Optional.of(appellantFamilyName));
+        when(asylumCase.read(LIST_CASE_HEARING_CENTRE, HearingCentre.class)).thenReturn(Optional.of(BIRMINGHAM));
+        when(asylumCase.read(ARIA_LISTING_REFERENCE, String.class)).thenReturn(Optional.of(listingReferenceNumber));
+        when((customerServicesProvider.getCustomerServicesTelephone())).thenReturn(customerServicesTelephone);
+        when((customerServicesProvider.getCustomerServicesEmail())).thenReturn(customerServicesEmail);
+
+        legalRepresentativeUpdateTribunalDecisionRule31PersonalisationEmail = new LegalRepresentativeUpdateTribunalDecisionRule31PersonalisationEmail(
+            legalRepresentativeUpdateTribunalDecisionRule31EmailTemplateId,
+            legalRepresentativeUpdateTribunalDecisionRule31AfterListingEmailTemplateId,
+            exUiFrontendUrl,
+            appealService,
+            customerServicesProvider
+        );
+    }
+
+    @Test
+    void should_return_given_template_id() {
+        when(appealService.isAppealListed(asylumCase)).thenReturn(true);
+
+        assertEquals(legalRepresentativeUpdateTribunalDecisionRule31AfterListingEmailTemplateId, legalRepresentativeUpdateTribunalDecisionRule31PersonalisationEmail.getTemplateId(asylumCase));
+    }
+
+    @Test
+    void should_return_given_reference_id() {
+        assertEquals(caseId + "_LEGAL_REPRESENTATIVE_UPDATE_TRIBUNAL_DECISION_RULE_31_EMAIL",
+                legalRepresentativeUpdateTribunalDecisionRule31PersonalisationEmail.getReferenceId(caseId));
+    }
+
+    @Test
+    void should_throw_exception_on_personalisation_when_case_is_null() {
+
+        assertThatThrownBy(() -> legalRepresentativeUpdateTribunalDecisionRule31PersonalisationEmail.getPersonalisation((Callback<AsylumCase>) null))
+            .isExactlyInstanceOf(NullPointerException.class)
+            .hasMessage("callback must not be null");
+    }
+
+    @Test
+    void should_return_personalisation_first_check_when_all_information_given() {
+
+        DynamicList dynamicList = new DynamicList(new Value("allowed", "Yes, change decision to Dismissed"),
```

Comment:
I think these values are incorrect, please check UpdateTribunalDecisionRule31MidEvent in ia-case-api